### PR TITLE
Feature(LSP): Completion and enhancement for GotoDefinition

### PR DESCRIPTION
**Pull request by He1pa**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y #297  

#### 2. What is the scope of this PR (e.g. component or file name):
LSP

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):
LSP Feature
+ Completion
    - import path
    - schema attr
    - builtin function(str)
    - definitions in pkg
+ GotoDefinition
  - schema attr
  - schema attr type

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [x] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [x] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: kclvm/tools/src/LSP/src/goto_def.rs [hunk 1]
```diff
@@ -180,3 +166,46 @@ fn goto_def_for_import(
     }
     positions_to_goto_def_resp(&positions)
 }
+
+fn find_def_of_schema_attr(
+    schema_expr: SchemaExpr,
```
Comment (Peefy) [inline kclvm/tools/src/LSP/src/goto_def.rs:171]: `ConfigExpr` also needs to be considered here. For example.

```python
schema Person:
    name: str
    data: Data

schema Data:
    id: int
    
person = Person {
    data.id = 1
    data: {
        id = 1
    }
    data: Data {
        id = 3
    }
}
```

File: kclvm/tools/src/LSP/src/completion.rs [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+                            Some(into_completion_items(&items).into())
+                        }
+                        None => None,
+                    }
+                }
+            },
+            None => None,
+        }
+    } else {
+        // todo
```
Comment (Peefy) [inline kclvm/tools/src/LSP/src/completion.rs:49]: This `todo` contains what?

File: kclvm/tools/src/LSP/src/completion.rs [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+/// Computes completions at the given position.
+pub(crate) fn completion(
+    trigger_character: Option<char>,
+    program: &Program,
+    pos: &KCLPos,
+    prog_scope: &ProgramScope,
+) -> Option<lsp_types::CompletionResponse> {
+    if let Some('.') = trigger_character{
+        match program.pos_to_stmt(&pos) {
```
Comment (Peefy) [inline kclvm/tools/src/LSP/src/completion.rs:32]: U can move this function to the `completion_dot` function?

File: kclvm/tools/src/LSP/src/completion.rs [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+fn completion_for_import(
+    stmt: &ImportStmt,
+    _pos: &KCLPos,
+    _prog_scope: &ProgramScope,
+    program: &Program,
+) -> Option<lsp_types::CompletionResponse> {
+    let mut items: IndexSet<String> = IndexSet::new();
+    let pkgpath = &stmt.path;
+    let real_path = Path::new(&program.root).join(pkgpath.replace('.', "/"));
```
Comment (Peefy) [inline kclvm/tools/src/LSP/src/completion.rs:62]: Note that for sep of different operating systems, U can use `std::path::MAIN_SEPARATOR`

File: kclvm/tools/src/LSP/src/completion.rs [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+    if real_path.is_dir() {
+        if let Ok(entries) = fs::read_dir(real_path) {
```
Comment (Peefy) [inline kclvm/tools/src/LSP/src/completion.rs:64]: use `get_kcl_files`?

File: kclvm/tools/src/LSP/src/completion.rs [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+                        }
+                    }
+                }
+            }
+        }
+    }
+    return Some(into_completion_items(&items).into());
+}
+
+fn get_completion_items(expr: Expr, prog_scope: &ProgramScope) -> IndexSet<String> {
```
Comment (Peefy) [inline kclvm/tools/src/LSP/src/completion.rs:92]: s/expr: Expr/expr: &Expr

File: kclvm/tools/src/LSP/src/completion.rs [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+    let mut items = IndexSet::new();
+    match expr {
+        Expr::Identifier(id) => {
+            let name = get_identifier_last_name(&id);
+            if !id.pkgpath.is_empty() {
+                // standard system module
+                if STANDARD_SYSTEM_MODULES.contains(&name.as_str())
```
Comment (Peefy) [inline kclvm/tools/src/LSP/src/completion.rs:99]: s/&name.as_str()/&name

File: kclvm/tools/src/LSP/src/completion.rs [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+                {
+                    items.extend(
+                        get_system_module_members(name.as_str())
+                            .iter()
+                            .map(|s| s.to_string()),
+                    )
+                }
+                // user module
+                if let Some(scope) = prog_scope.scope_map.get(&id.pkgpath) {
```
Comment (Peefy) [inline kclvm/tools/src/LSP/src/completion.rs:108]: Lost plugin module?

File: kclvm/tools/src/LSP/src/completion.rs (hunk 5 above)
Comment (He1pa) [reply kclvm/tools/src/LSP/src/completion.rs:64]: When processing completion in the import statement, in addition to the kcl file, the directory also be completed

**Review by Peefy — APPROVED** [review]
LGTM

**Pull request merged**